### PR TITLE
Update selinux readme

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

Update selinux readme

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: scripts/selinux/README.md [hunk 1]
```diff
@@ -36,16 +44,42 @@ sudo restorecon -R /etc/nginx-agent
 ```
 
 ## Updating existing policy
-Copy the folder `scripts/selinux` over to your rhel server.
+Copy the folder `scripts/selinux` over to your rhel 8 server.
```
Comment (aphralG) [inline scripts/selinux/README.md:47]: would it be worth mentioning that you need to change the file permission on nginx_agent.sh ?

File: scripts/selinux/README.md [hunk 2]
```diff
@@ -36,16 +44,42 @@ sudo restorecon -R /etc/nginx-agent
...
 Then run the following command to update the policy:
 ```
 sudo ./nginx_agent.sh --update
 ```
+Then copy the `nginx_agent.te` and `nginx_agent.pp` files back and create a PR with the changes.
```
Comment (aphralG) [inline scripts/selinux/README.md:52]: Is it worth adding to check the policy doesn't need further updates with
`sudo ausearch -m avc --raw -se nginx_agent | audit2allow -R`  before you copy back the files and make a PR ?

Comment (dhurley) [reply scripts/selinux/README.md:52]: The nginx_agent.sh script already handles that. See here https://github.com/nginx/agent/blob/main/scripts/selinux/nginx_agent.sh#L14 & here https://github.com/nginx/agent/blob/main/scripts/selinux/nginx_agent.sh#L17

File: scripts/selinux/README.md (hunk 1 above)
Comment (dhurley) [reply scripts/selinux/README.md:47]: Good point. I'll add that

File: scripts/selinux/README.md (hunk 2 above)
Comment (aphralG) [reply scripts/selinux/README.md:52]: ok was just thinking that we ran it to check last time so was wondering if it needed to be in the doc

**Review by Dean-Coakley — APPROVED** [review]

**Review by aphralG — APPROVED** [review]

**Review by oliveromahony — APPROVED** [review]

File: scripts/selinux/README.md [hunk 3]
```diff
@@ -36,10 +50,61 @@ sudo restorecon -R /etc/nginx-agent
...
+Check for errors by using the `ausearch` command:
+```
+sudo ausearch -m AVC,USER_AVC,SELINUX_ERR,USER_SELINUX_ERR --raw -se nginx_agent -ts recent
+```
+Generate new rule based on the errors by using `audit2allow`:
+```
+sudo ausearch -m AVC,USER_AVC,SELINUX_ERR,USER_SELINUX_ERR --raw -se nms -ts recent | audit2allow
+```
+
+Copy the file `scripts/selinux/nginx_agent.te` to a centos 7 machine and update it with the output from the `audit2allow` command.
```
Comment (aphralG) [inline scripts/selinux/README.md:62]: Is there a way to make it more clear that we are not saying to only run the audit2allow command on centos 7 and then copy the file. But to run the audit2allow command on 8 & 9 etc copy the output of that to the nginx_agent.te file and then copy that to centos 7  to build ?

Comment (dhurley) [reply scripts/selinux/README.md:62]: yeah let me add a note there to make it more clear

**Review by aphralG — APPROVED** [review]

File: scripts/selinux/README.md [hunk 4]
```diff
@@ -36,10 +50,62 @@ sudo restorecon -R /etc/nginx-agent
...
+
+Copy the `scripts/selinux/nginx_agent.te` file to a Centos 7 machine and build a new `nginx_agent.pp` file by running the following command:
+```
+make -f /usr/share/selinux/devel/Makefile nginx_agent.pp
+```
+**[NOTE: The policy has to be built on a Centos 7 machine. If it is built on a different OS like RHEL 8/9 then we will encounter this issue [Policy version does not match](#policy-version-does-not-match) when installing it on different an older OS like Centos 7. Even if the audit2allow command was run on a RHEL 8/9 machine the updates to the policy need to be made on a Centos 7 machine.]**
+
+Install the policy by following the steps here [Install NGINX Agent Policy](#install-nginx-agent-policy)
+
+Then create a PR with the chnages made to the `nginx_agent.te` and `nginx_agent.pp` files.
```
Comment (Dean-Coakley) [inline scripts/selinux/README.md:72]:
```suggestion
Then create a PR with the changes made to the `nginx_agent.te` and `nginx_agent.pp` files.
```

File: scripts/selinux/README.md [hunk 5]
```diff
@@ -36,10 +50,62 @@ sudo restorecon -R /etc/nginx-agent
...
+sudo ausearch -m AVC,USER_AVC,SELINUX_ERR,USER_SELINUX_ERR --raw -se nms -ts recent | audit2allow
+```
+
+Update the `scripts/selinux/nginx_agent.te` file with the output from the `audit2allow` command.
+
+Copy the `scripts/selinux/nginx_agent.te` file to a Centos 7 machine and build a new `nginx_agent.pp` file by running the following command:
+```
+make -f /usr/share/selinux/devel/Makefile nginx_agent.pp
+```
+**[NOTE: The policy has to be built on a Centos 7 machine. If it is built on a different OS like RHEL 8/9 then we will encounter this issue [Policy version does not match](#policy-version-does-not-match) when installing it on different an older OS like Centos 7. Even if the audit2allow command was run on a RHEL 8/9 machine the updates to the policy need to be made on a Centos 7 machine.]**
```
Comment (Dean-Coakley) [inline scripts/selinux/README.md:68]:
```suggestion
**[NOTE: The policy has to be built on a Centos 7 machine. If it is built on a different OS like RHEL 8/9 then we will encounter this issue [Policy version does not match](#policy-version-does-not-match) when installing it on different an older OS like Centos 7. Even if the `audit2allow` command was run on a RHEL 8/9 machine the updates to the policy need to be made on a Centos 7 machine.]**
```

**Review by Dean-Coakley — APPROVED** [review]

**Pull request merged**